File: Data Structures & Algorithms/binary-search/submission-0.cs
public class Solution {
    public int Search(int[] nums, int target) {
        int left = 0;
        int right = nums.Length - 1;
        while (left <= right) {
            int middle = left + (right - left) / 2;

            if (target > nums[middle]) {
                left = middle + 1;
            } else if (target < nums[middle]) {
                right = middle - 1;
            } else if (target == nums[middle]) {
                return middle;
            }
        }
        
        return -1;
    }
}

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        vector<int> newNums;
        if (nums.size() == 0)
            return 0;
        newNums.push_back(nums[0]);
        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] == nums[i - 1]) {
                continue ;
            }
            newNums.push_back(nums[i]);
        }
        nums = newNums;
        return newNums.size();
    }
};